src/main/java/uk/gov/digital/ho/pttg/application/namematching/candidates/AliasSurnameCombinationsFunctions.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.digital.ho.pttg.application.namematching.candidates;
+
+import uk.gov.digital.ho.pttg.application.namematching.CandidateName;
+import uk.gov.digital.ho.pttg.application.namematching.InputNames;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.reverse;
+
+final class AliasSurnameCombinationsFunctions {
+
+    private AliasSurnameCombinationsFunctions() {
+        throw new UnsupportedOperationException("Helper class for AliasSurnameCombinationsFunctions containing only static methods - no need to instantiate.");
+    }
+
+    static List<String> removeName(String nameToRemove, List<String> names) {
+        return names.stream()
+                .filter(name -> !name.equals(nameToRemove))
+                .collect(Collectors.toList());
+    }
+
+    static List<CandidateName> nonAliasFirstAliasLastCombinations(List<String> nonAliasNames, List<String> aliasSurnames) {
+        List<CandidateName> candidateNames = new ArrayList<>();
+
+        List<String> reversedAliasSurnames = new ArrayList<>(aliasSurnames);
+        reverse(reversedAliasSurnames);
```

Comment:
Is this method working from a side effect?  Feels like it should return a new list.

Reply:
Yes it's Collections.reverse. I probably should put the qualifier in to make this clear. I don't like it but I have yet to discover a more fluent way to reverse a list. Definitely open to suggestions!